src/@awesome-cordova-plugins/plugins/purchases/index.ts

```diff
@@ -291,7 +436,7 @@ export class Purchases extends AwesomeCordovaNativePlugin {
    * @param {PurchasesConfiguration} Object containing configuration parameters
    */
   @Cordova({ sync: true })
-  configureWith({ apiKey, appUserID, observerMode, userDefaultsSuiteName, useAmazon }: PurchasesConfiguration): void {}
+  configureWith({ apiKey, appUserID, observerMode, userDefaultsSuiteName, useAmazon, usesStoreKit2IfAvailable, shouldShowInAppMessagesAutomatically }: PurchasesConfiguration): void {}
```

Comment:
Hmm I was wondering, even though these accept undefined in `PurchasesConfiguration`, do we need to set a default value here? Though I think the better solution would be to just pass a `PurchasesConfiguration` object, instead of redefining the fields here.

Reply:
[not a ts person here so don't take my word for any of this, but]

I guess the question is whether passing it as an object just works in awesome-cordova-plugins, though, since accessing the variables would look a little different: using destructuring like this, you can access the parameters (`apiKey`, `appUserID`, etc) directly, whereas if you pass in something like a `configuration` object, you'd access as `config.apiKey`, `config.appUserID`, etc. 

Since we don't provide an implementation and rely on awesome-cordova-plugins to do the magic and pass it onto cordova, we should make sure that whatever we do actually works correctly there

Reply:
Ah true... So yeah, if we can't just `PurchasesConfiguration`, I think we might still need to add default values to these new parameters to try to avoid the breaking change if possible, similar to how we do in the [cordova sdk](https://github.com/RevenueCat/cordova-plugin-purchases/blob/main/src/plugin/plugin.ts#L1020).

Also, looking at Cordova, the order of the fields is not the same, don't think that should cause issues but I'm not that familiar with the mapping done from here to the Cordova SDK so I wanted to mention it.

Reply:
I will try passing the `PurchasesConfiguration` object and report back

Reply:
Hmm yeah, makes sense... I think since some of the parameters can be undefined, that should actually be the default. That way we can rely on the default set in the native SDK.

But in any case, if we want to mimick cordova and add the defaults here, I think it should be ok as well. Just want to make sure it will work fine even with the addition of these 2 new fields considering we don't have API tests here.

Reply:
Yes. I tried to test it yesterday but had some issues building the plugin. Today I realized there's a broken plugin in awesome-cordova-plugins and removing it makes it work. Hopefully I can test it today without spending too much time on it 😅

Reply:
I finally managed to test this and it looks like even if we don't define defaults here, it will respect whatever cordova has defined. Look at this:

<img width="1310" alt="Screenshot 2023-11-03 at 16 37 48" src="https://github.com/RevenueCat/ionic-native/assets/664544/952485d5-7d90-4987-b0e9-136b9e936f0f">

`shouldShowInAppMessagesAutomatically` is set to `true`, which is the default in Cordova.

Reply:
Nice thanks for checking that!